File: game-server/src/main/java/net/dodian/jobs/impl/ActionProcessor.java
package net.dodian.jobs.impl;

import net.dodian.uber.game.runtime.queue.QueueTaskService;
import net.dodian.uber.game.event.GameEventBus;
import net.dodian.uber.game.event.events.WorldTickEvent;
import net.dodian.uber.game.model.entity.player.PlayerHandler;

/**
 * Processes queued game actions/events in tick-order.
 */
public class ActionProcessor implements Runnable {
    @Override
    public void run() {
        QueueTaskService.processDue();
        GameEventBus.INSTANCE.post(new WorldTickEvent(PlayerHandler.cycle));
    }
}

File: game-server/src/main/kotlin/net/dodian/uber/game/engine/processing/ActionProcessor.kt
package net.dodian.uber.game.engine.processing

import net.dodian.uber.game.engine.scheduler.QueueTaskService
import net.dodian.uber.game.engine.event.GameEventBus
import net.dodian.uber.game.events.WorldTickEvent
import net.dodian.uber.game.engine.systems.world.player.PlayerRegistry

/**
 * Processes queued game actions/events in tick-order.
 */
class ActionProcessor : Runnable {
    override fun run() {
        QueueTaskService.processDue()
        GameEventBus.post(WorldTickEvent(PlayerRegistry.cycle))
    }
}
